storage/mysql/log_storage.go

```diff
@@ -569,32 +568,21 @@ func (t *logTreeTX) UpdateSequencedLeaves(ctx context.Context, leaves []*trillia
 
 // removeSequencedLeaves removes the passed in leaves slice (which may be
 // modified as part of the operation).
-func (t *logTreeTX) removeSequencedLeaves(ctx context.Context, leaves []*trillian.LogLeaf) error {
-	// Delete in order of the hash values in the leaves.
-	sort.Sort(byLeafIdentityHash(leaves))
-
+func (t *logTreeTX) removeSequencedLeaves(ctx context.Context, leaves []*dequeuedLeaf) error {
+	// Don't need to re-sort because the query ordered by leaf hash. If that changes because
+	// the query is expensive then the sort will need to be done here.
 	tmpl, err := t.ls.getDeleteUnsequencedStmt(ctx, len(leaves))
```

Comment:
There's no placeholder on deleteUnsequencedSQL anymore. Do we still need getDeleteUnsequencedStmt? If yes, could we remove the `num` parameter, as it's a bit misleading now?